src/Symfony/Component/Validator/Mapping/PreprocessingMetadataInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Mapping;
+
+/**
+ * Interface for validation metadata that pre-processes values.
+ *
+ * Metadata implementing this interface is able to pre-process values before
+ * they are passed on the constraint validators.
+ *
+ * @since  2.7
+ *
+ * @author Wolfgang Ziegler <fago@wolfgangziegler.net>
+ *
+ * @see CascadingStrategy
+ * @see TraversalStrategy
+ */
+interface PreprocessingMetadataInterface extends MetadataInterface
+{
+    /**
+     * Pre-processes the value before validation.
+     *
+     * @param mixed $value The value to be validated.
+     */
+    public function preprocessValue(&$value);
```

Comment:
Please don't use it by reference. It will force introducing reference mismatches (see http://jpauli.github.io/2014/06/27/references-mismatch.html for the explanation) and it makes the API harder to understand. Returning a value is much better.